### PR TITLE
Add PowerShell install module

### PR DESCRIPTION
**Pull request by RassK**
## What

Adds PowerShell install module.

### CODE REVIEW

File: Splunk.OTel.DotNet.psm1 [hunk 1]
```diff
@@ -0,0 +1,375 @@
+﻿#Requires -RunAsAdministrator
+
+function Get-Current-InstallDir() {
+    return [System.Environment]::GetEnvironmentVariable("OTEL_DOTNET_AUTO_INSTALL_DIR", [System.EnvironmentVariableTarget]::Machine)
```
Comment (pellared) [inline Splunk.OTel.DotNet.psm1:4]: Minor: maybe `SPLUNK_OTEL_DOTNET_AUTO_INSTALL_DIR` in order to offer side-by-side installation? 🤔

**Review by pellared — APPROVED** [review]

**Comment by pellared** [comment]
Good to merge when the tests are green

**Review by Kielek — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -74,7 +74,7 @@ jobs:
           Remove-Item $log_path
           Unregister-OpenTelemetryForCurrentSession
           dotnet help
-          If (Test-Path $log_path) { throw "Log file exist. Instrumentation uninstall failed." }
+          If (Test-Path $log_path) { throw "Log file exists. Instrumentation uninstall failed." }
```
Comment (pellared) [inline .github/workflows/ci.yml:77]: Maybe we should also test the uninstall function?

Comment (RassK) [reply .github/workflows/ci.yml:77]: testing if the core files were removed?

Comment (pellared) [reply .github/workflows/ci.yml:77]: yup I am just not sure if `Remove-Item -LiteralPath $InstallDir -Force -Recurse` works 😅

Comment (pellared) [reply .github/workflows/ci.yml:77]: Here is the reason of my doubts https://github.com/signalfx/splunk-otel-dotnet/pull/41/commits/aa317bdbc1f48eb95a9db5bf5095ae3f75ba111f

Comment (RassK) [reply .github/workflows/ci.yml:77]: removing dir vs contents, but still no issues, can add.

**Review by pellared — APPROVED** [review]

**Pull request merged**